src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1071,4 +1072,152 @@ function ($a) {
+                                                        ->integerNode('critical')->defaultNull()->end()
+                                                    ->end()
+                                                ->end()
+                                            ->end()
+                                            ->validate()
+                                                ->ifTrue(function ($config) {
+                                                    return 'constant' === $config['retry_strategy'] && !array_key_exists('max', $config['retry_strategy_options']);
+                                                })
+                                                ->thenInvalid('"max" of "retry_strategy_options" must be set for constant retry strategy.')
+                                            ->end()
+                                            ->validate()
+                                                ->ifTrue(function ($config) {
+                                                    return 'constant' === $config['retry_strategy'] && !array_key_exists('time', $config['retry_strategy_options']);
+                                                })
+                                                ->thenInvalid('"time" of "retry_strategy_options" must be set for constant retry strategy.')
+                                            ->end()
+                                            ->validate()
+                                                ->ifTrue(function ($config) {
+                                                    return 'exponential' === $config['retry_strategy'] && !array_key_exists('max', $config['retry_strategy_options']);
+                                                })
+                                                ->thenInvalid('"max" of "retry_strategy_options" must be set for exponential retry strategy.')
+                                            ->end()
+                                            ->validate()
+                                                ->ifTrue(function ($config) {
+                                                    return 'exponential' === $config['retry_strategy'] && !array_key_exists('offset', $config['retry_strategy_options']);
+                                                })
+                                                ->thenInvalid('"offset" of "retry_strategy_options" must be set for exponential retry strategy.')
+                                            ->end()
+                                        ->end()
+                                    ->end()
+                                ->end()
+                            ->end()
+                        ->end()
+                        ->scalarNode('default_connection')
+                            ->cannotBeEmpty()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+    }
+
+    private function fixXmlArguments($v)
```

Comment:
What's exactly fixed by this method? Can't you simply have an XSD that allows any kind of data like the rest of the "free options"? 🤔